flink/src/main/java/org/apache/iceberg/flink/actions/MigrateAction.java

```diff
@@ -0,0 +1,374 @@
+
+    DataStream<PartitionAndLocation> dataStream = env.fromCollection(inputs);
+    MigrateMap migrateMap = new MigrateMap(spec, nameMapping, fileFormat, metricsConfig);
+    DataStream<List<DataFile>> ds = dataStream.map(migrateMap).setParallelism(parallelism);
+    return Lists.newArrayList(ds.executeAndCollect("migrate table :" + hiveTableName)).stream()
+        .flatMap(Collection::stream).collect(Collectors.toList());
+  }
+
+  private static class MigrateMap implements MapFunction<PartitionAndLocation, List<DataFile>> {
+    private final PartitionSpec spec;
+    private final String nameMappingString;
+    private final FileFormat fileFormat;
+    private final MetricsConfig metricsConfig;
+
+    MigrateMap(PartitionSpec spec, String nameMapping, FileFormat fileFormat, MetricsConfig metricsConfig) {
+      this.spec = spec;
+      this.nameMappingString = nameMapping;
+      this.fileFormat = fileFormat;
+      this.metricsConfig = metricsConfig;
+    }
+
+    @Override
+    public List<DataFile> map(PartitionAndLocation map) {
+      Map<String, String> partitions = map.getMap();
+      String location = map.getLocation();
+      Configuration conf = HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
+      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      List<DataFile> files;
+      switch (fileFormat) {
+        case PARQUET:
+          files = listParquetPartition(partitions, location, spec, conf, metricsConfig, nameMapping);
+          break;
+
+        case ORC:
+          files = listOrcPartition(partitions, location, spec, conf, metricsConfig, nameMapping);
+          break;
+
+        default:
+          throw new UnsupportedOperationException("Unsupported file format");
+      }
+
+      return files;
+    }
+
+    private List<DataFile> listOrcPartition(Map<String, String> partitionPath, String partitionUri,
+                                            PartitionSpec partitionSpec, Configuration conf,
+                                            MetricsConfig metricsSpec, NameMapping mapping) {
+      try {
+        Path partition = new Path(partitionUri);
+        FileSystem fs = partition.getFileSystem(conf);
+
+        return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+            .filter(FileStatus::isFile)
+            .map(stat -> {
+              Metrics metrics = OrcMetrics.fromInputFile(HadoopInputFile.fromPath(stat.getPath(), conf),
+                  metricsSpec, mapping);
+              String partitionKey = partitionSpec.fields().stream()
+                  .map(PartitionField::name)
+                  .map(name -> String.format("%s=%s", name, partitionPath.get(name)))
+                  .collect(Collectors.joining("/"));
+
+              return DataFiles.builder(partitionSpec)
+                  .withPath(stat.getPath().toString())
+                  .withFormat(FileFormat.ORC.name())
```

Comment:
nit: there is a `withFormat` overload that takes the enum directly. we don't have to call `.name()`